CMakeLists.txt

```diff
@@ -57,6 +57,7 @@ check_library_exists(m pow "" HAVE_LIBM)
 check_include_file("dirent.h" HAVE_DIRENT_H)
 check_symbol_exists(strcasecmp "strings.h" HAVE_DECL_STRCASECMP)
 check_symbol_exists(strncasecmp "strings.h" HAVE_DECL_STRNCASECMP)
+check_symbol_exists(CREATE_WAITABLE_TIMER_HIGH_RESOLUTION "windows.h" HAVE_HIGH_RES_TIMER)
```

Comment:
`check_symbol_exists` may not be enough. @MrAlaux Could you please build this PR on your system?

Reply:
"If the symbol is a type, enum value, or intrinsic it will not be recognized (consider using [CheckTypeSize](https://cmake.org/cmake/help/latest/module/CheckTypeSize.html#module:CheckTypeSize) or [CheckSourceCompiles](https://cmake.org/cmake/help/latest/module/CheckSourceCompiles.html#module:CheckSourceCompiles))."
https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html

Reply:
It's a macro in Mingw-W64:

```
D:/msys64/mingw64/include/synchapi.h
131:#define CREATE_WAITABLE_TIMER_HIGH_RESOLUTION 0x2
```

Reply:
It's a macro:
```c
#define CREATE_WAITABLE_TIMER_MANUAL_RESET  0x00000001
#if (_WIN32_WINNT >= _NT_TARGET_VERSION_WIN10_RS4)
#define CREATE_WAITABLE_TIMER_HIGH_RESOLUTION 0x00000002
#endif
```
However, I'm not sure how CMake checks for symbols. Perhaps it just searches for strings, so we need `check_c_source_compiles`.

Reply:
From the same docs:
"If the header files define the symbol as a macro it is considered available and assumed to work."

Reply:
Using `check_c_source_compiles` is probably playing extra safe.